node/silkworm/stagedsync/util.cpp

```diff
@@ -43,8 +43,19 @@ std::pair<Bytes, Bytes> convert_to_db_format(const ByteView& key, const ByteView
     return {a, b};
 }
 
+/*
+ * Truncate a table by starting from a certain key, either up or down.
+ * If reverse is set to false, then every key either higher or equal to starting key will be erased.
+ * If reverse is set to true, then every key either lower or equal to starting key will be erased.
```

Comment:
Well, this is not what the function does, is it? If reverse is set to true, then every key strictly lower than `starting_key` will be erased, not `<= starting_key`.

I'd also remove or rephrase the NOTE. The fact that we use `cursor. lower_bound()` is just an implementation detail, not relevant to the API. We could implement this function differently and achieve `<= starting_key` for the reverse case with something like
```
if (current && current.key == db::to_slice(starting_key)) {
    table.erase(/*whole_multivalue=*/true);
}
```

Reply:
ok, i fixed the comments as you said